File: app/src/main/java/powerball/apps/jacs/powerball/MyNumbers.java
package powerball.apps.jacs.powerball;

import android.content.Context;
import android.content.DialogInterface;
import android.graphics.Color;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.AlertDialog;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.NumberPicker;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

import java.util.ArrayList;


public class MyNumbers extends Fragment {
    public TextView powerBall,megaMillions;
    public ArrayList<MyTicket> powerNumbers,megaNumbers;
    public RecyclerView power,mega;
    public MyTicketAdapter powerAdapter,megaAdapter;
    public MyNumbers() {
        // Required empty public constructor
    }


    public static MyNumbers newInstance() {
        MyNumbers fragment = new MyNumbers();

        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_my_numbers, container, false);
    }



    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

    }

    @Override
    public void onDetach() {
        super.onDetach();

    }

    @Override
    public void onStart() {
        super.onStart();
        instantiateViews();
        getMyTickets();
        populateRecyclerViews();
        setClickListeners();
    }
    public void instantiateViews(){
        View power1 = (View)getView().findViewById(R.id.include);
        View mega1 = (View)getView().findViewById(R.id.include1);
        powerBall = (TextView)power1.findViewById(R.id.win);
        megaMillions = (TextView)mega1.findViewById(R.id.winm);
        power = (RecyclerView)getView().findViewById(R.id.powerRecycle);
        mega = (RecyclerView)getView().findViewById(R.id.megaRecycle);
    }
    public void getMyTickets(){
        powerNumbers = SharedPrefHelper.getMyTickets(getActivity(),"powerball");
        megaNumbers = SharedPrefHelper.getMyTickets(getActivity(),"megamillions");
    }
    public void setMyTickets(){
        SharedPrefHelper.setMyTickets(getActivity(),powerNumbers,"powerball");
        SharedPrefHelper.setMyTickets(getActivity(),megaNumbers,"megamillions");
    }
    public void populateRecyclerViews(){
        //power.setHasFixedSize(true);
        LinearLayoutManager llm = new LinearLayoutManager(getActivity());
        power.setLayoutManager(llm);
        powerAdapter = new MyTicketAdapter(this,powerNumbers,"powerball");
        power.setAdapter(powerAdapter);
        powerAdapter.setOnRemoveListener(new OnRemoveListener() {
            @Override
            public void onRemove(int position) {

                    powerNumbers.remove(position);
                    Log.d("adapter",powerNumbers.size()+"");
                    //power.removeViewAt(position);
                    power.removeAllViews();
                    //powerAdapter.notifyItemRemoved(position);
                    //powerAdapter.notifyItemRangeChanged(position, powerNumbers.size());
                    powerAdapter.notifyDataSetChanged();
                    SharedPrefHelper.setMyTickets(getActivity(),powerNumbers,"powerball");
            }
        });

        //mega.setHasFixedSize(true);
        LinearLayoutManager llm1 = new LinearLayoutManager(getActivity());
        mega.setLayoutManager(llm1);
        megaAdapter = new MyTicketAdapter(this,megaNumbers,"megamillions");

        mega.setAdapter(megaAdapter);
        megaAdapter.setOnRemoveListener(new OnRemoveListener() {
            @Override
            public void onRemove(int position) {
                megaNumbers.remove(position);
                mega.removeAllViews();
                megaAdapter.notifyDataSetChanged();
                SharedPrefHelper.setMyTickets(getActivity(),megaNumbers,"megamillions");
            }
        });
    }
    public void removeView(String str,int position){
        if(str.equals("powerball")){
            power.removeViewAt(position);
        }
    }
    public void setClickListeners(){
        powerBall.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity());
// ...Irrelevant code for customizing the buttons and title
                LayoutInflater inflater = getActivity().getLayoutInflater();
                final View dialogView = inflater.inflate(R.layout.ticket_selector, null);
                final NumberPicker one = (NumberPicker)dialogView.findViewById(R.id.one);
                one.setMinValue(1);
                one.setMaxValue(69);
                final NumberPicker two = (NumberPicker)dialogView.findViewById(R.id.two);
                two.setMinValue(1);
                two.setMaxValue(69);
                final NumberPicker three = (NumberPicker)dialogView.findViewById(R.id.three);
                three.setMinValue(1);
                three.setMaxValue(69);
                final NumberPicker four = (NumberPicker)dialogView.findViewById(R.id.four);
                four.setMinValue(1);
                four.setMaxValue(69);
                final NumberPicker five = (NumberPicker)dialogView.findViewById(R.id.five);
                five.setMinValue(1);
                five.setMaxValue(69);
                final NumberPicker six = (NumberPicker)dialogView.findViewById(R.id.six);
                six.setMinValue(1);
                six.setMaxValue(26);
                final RadioGroup group = (RadioGroup)dialogView.findViewById(R.id.group);

                dialogBuilder.setView(dialogView);




                final AlertDialog alertDialog = dialogBuilder.create();
                alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "Cancel", null, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        alertDialog.dismiss();
                    }
                });
                alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "Add", null, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        //powerNumbers = SharedPrefHelper.getMyTickets(getActivity(),"powerball");
                        //powerNumbers = powerAdapter.mDataset;
                        String ticket = one.getValue() + " " + two.getValue() + " " + three.getValue() +
                                " " + four.getValue() + " " + five.getValue() + " " + six.getValue();
                        int mul = group.getCheckedRadioButtonId();
                        RadioButton but = dialogView.findViewById(mul);
                        String multi = but.getText().toString();
                        boolean ticketMultiplier = false;
                        if(multi.equals("Yes")){
                            ticketMultiplier = true;
                        }
                        MyTicket ticket1 = new MyTicket();
                        ticket1.ticket = ticket;
                        ticket1.multi = ticketMultiplier;
                        powerNumbers.add(ticket1);

                        Log.d("adapter",powerNumbers.size()+"");
                        powerAdapter.notifyItemInserted(powerNumbers.size() - 1);
                        SharedPrefHelper.setMyTickets(getActivity(),powerNumbers,"powerball");
                        alertDialog.dismiss();

                    }
                });
                alertDialog.show();
            }
        });
        megaMillions.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity());
// ...Irrelevant code for customizing the buttons and title
                LayoutInflater inflater = getActivity().getLayoutInflater();
                final View dialogView = inflater.inflate(R.layout.ticket_selector, null);
                final NumberPicker one = (NumberPicker)dialogView.findViewById(R.id.one);
                one.setMinValue(1);
                one.setMaxValue(69);
                final NumberPicker two = (NumberPicker)dialogView.findViewById(R.id.two);
                two.setMinValue(1);
                two.setMaxValue(69);
                final NumberPicker three = (NumberPicker)dialogView.findViewById(R.id.three);
                three.setMinValue(1);
                three.setMaxValue(69);
                final NumberPicker four = (NumberPicker)dialogView.findViewById(R.id.four);
                four.setMinValue(1);
                four.setMaxValue(69);
                final NumberPicker five = (NumberPicker)dialogView.findViewById(R.id.five);
                five.setMinValue(1);
                five.setMaxValue(69);
                final NumberPicker six = (NumberPicker)dialogView.findViewById(R.id.six);
                six.setMinValue(1);
                six.setMaxValue(26);
                six.setBackgroundColor(Color.YELLOW);
                final RadioGroup group = (RadioGroup)dialogView.findViewById(R.id.group);

                dialogBuilder.setView(dialogView);




                final AlertDialog alertDialog = dialogBuilder.create();
                alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "Cancel", null, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        alertDialog.dismiss();
                    }
                });
                alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "Add", null, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        //megaNumbers.clear();
                        //megaNumbers = SharedPrefHelper.getMyTickets(getActivity(),"megamillions");
                       // megaNumbers = new ArrayList<MyTicket>(megaAdapter.mDataset);
                        String ticket = one.getValue() + " " + two.getValue() + " " + three.getValue() +
                                " " + four.getValue() + " " + five.getValue() + " " + six.getValue();
                        int mul = group.getCheckedRadioButtonId();
                        RadioButton but = dialogView.findViewById(mul);
                        String multi = but.getText().toString();
                        boolean ticketMultiplier = false;
                        if(multi.equals("Yes")){
                            ticketMultiplier = true;
                        }
                        MyTicket ticket1 = new MyTicket();
                        ticket1.ticket = ticket;
                        ticket1.multi = ticketMultiplier;
                        megaNumbers.add(ticket1);
                        megaAdapter.notifyItemInserted(megaNumbers.size() - 1);
                        //megaAdapter.notifyDataSetChanged();
                        SharedPrefHelper.setMyTickets(getActivity(),megaNumbers,"megamillions");
                        alertDialog.dismiss();

                    }
                });
                alertDialog.show();
            }
        });
    }
}

File: app/src/main/java/powerball/apps/jacs/powerball/MyNumbers.kt
package powerball.apps.jacs.powerball

import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AlertDialog
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.NumberPicker
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.TextView
import java.util.*

class MyNumbers : Fragment() {
    var powerBall: TextView? = null
    var megaMillions: TextView? = null
    var powerNumbers: ArrayList<MyTicket>? = null
    var megaNumbers: ArrayList<MyTicket>? = null
    var power: RecyclerView? = null
    var mega: RecyclerView? = null
    var powerAdapter: MyTicketAdapter? = null
    var megaAdapter: MyTicketAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_my_numbers, container, false)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
    }

    override fun onDetach() {
        super.onDetach()
    }

    override fun onStart() {
        super.onStart()
        instantiateViews()
        myTickets
        populateRecyclerViews()
        setClickListeners()
    }

    fun instantiateViews() {
        val power1 = view!!.findViewById(R.id.include) as View
        val mega1 = view!!.findViewById(R.id.include1) as View
        powerBall = power1.findViewById<View>(R.id.win) as TextView
        megaMillions = mega1.findViewById<View>(R.id.winm) as TextView
        power = view!!.findViewById<View>(R.id.powerRecycle) as RecyclerView
        mega = view!!.findViewById<View>(R.id.megaRecycle) as RecyclerView
    }

    val myTickets: Unit
        get() {
            powerNumbers = SharedPrefHelper.getMyTickets(this.context!!, "powerball")
            megaNumbers = SharedPrefHelper.getMyTickets(this.context!!, "megamillions")
        }

    fun setMyTickets() {
        SharedPrefHelper.setMyTickets(this.context!!, powerNumbers, "powerball")
        SharedPrefHelper.setMyTickets(this.context!!, megaNumbers, "megamillions")
    }

    fun populateRecyclerViews() {
        //power.setHasFixedSize(true);
        val llm = LinearLayoutManager(activity)
        power!!.layoutManager = llm
        powerAdapter = MyTicketAdapter(this, powerNumbers!!, "powerball")
        power!!.adapter = powerAdapter
        val onRemoveListener = object: OnRemoveListener{
            override fun onRemove(position: Int) {
                powerNumbers!!.removeAt(position)
                Log.d("adapter", powerNumbers!!.size.toString() + "")
                //power.removeViewAt(position);
                power!!.removeAllViews()
                //powerAdapter.notifyItemRemoved(position);
                //powerAdapter.notifyItemRangeChanged(position, powerNumbers.size());
                powerAdapter!!.notifyDataSetChanged()
                SharedPrefHelper.setMyTickets(this@MyNumbers.context!!, powerNumbers, "powerball")
            }

        }
        powerAdapter!!.listener = onRemoveListener

        //mega.setHasFixedSize(true);
        val llm1 = LinearLayoutManager(activity)
        mega!!.layoutManager = llm1
        megaAdapter = MyTicketAdapter(this, megaNumbers!!, "megamillions")
        mega!!.adapter = megaAdapter
        val listener = object : OnRemoveListener{
            override fun onRemove(position: Int) {
                megaNumbers!!.removeAt(position)
                mega!!.removeAllViews()
                megaAdapter!!.notifyDataSetChanged()
                SharedPrefHelper.setMyTickets(this@MyNumbers.context!!, megaNumbers, "megamillions")
            }

        }
        megaAdapter!!.listener = listener
    }

    fun removeView(str: String, position: Int) {
        if (str == "powerball") {
            power!!.removeViewAt(position)
        }
    }

    fun setClickListeners() {
        powerBall!!.setOnClickListener {
            val dialogBuilder = AlertDialog.Builder(activity!!)
            // ...Irrelevant code for customizing the buttons and title
            val inflater = activity!!.layoutInflater
            val dialogView = inflater.inflate(R.layout.ticket_selector, null)
            val one = dialogView.findViewById<View>(R.id.one) as NumberPicker
            one.minValue = 1
            one.maxValue = 69
            val two = dialogView.findViewById<View>(R.id.two) as NumberPicker
            two.minValue = 1
            two.maxValue = 69
            val three = dialogView.findViewById<View>(R.id.three) as NumberPicker
            three.minValue = 1
            three.maxValue = 69
            val four = dialogView.findViewById<View>(R.id.four) as NumberPicker
            four.minValue = 1
            four.maxValue = 69
            val five = dialogView.findViewById<View>(R.id.five) as NumberPicker
            five.minValue = 1
            five.maxValue = 69
            val six = dialogView.findViewById<View>(R.id.six) as NumberPicker
            six.minValue = 1
            six.maxValue = 26
            val group = dialogView.findViewById<View>(R.id.group) as RadioGroup
            dialogBuilder.setView(dialogView)
            val alertDialog = dialogBuilder.create()
            alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "Cancel", null) { dialog, which -> alertDialog.dismiss() }
            alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "Add", null) { dialog, which -> //powerNumbers = SharedPrefHelper.getMyTickets(getActivity(),"powerball");
                //powerNumbers = powerAdapter.mDataset;
                val ticket = one.value.toString() + " " + two.value + " " + three.value +
                        " " + four.value + " " + five.value + " " + six.value
                val mul = group.checkedRadioButtonId
                val but = dialogView.findViewById<RadioButton>(mul)
                val multi = but.text.toString()
                var ticketMultiplier = false
                if (multi == "Yes") {
                    ticketMultiplier = true
                }
                val ticket1 = MyTicket()
                ticket1.ticket = ticket
                ticket1.multi = ticketMultiplier
                powerNumbers!!.add(ticket1)
                Log.d("adapter", powerNumbers!!.size.toString() + "")
                powerAdapter!!.notifyItemInserted(powerNumbers!!.size - 1)
                SharedPrefHelper.setMyTickets(this.context!!, powerNumbers, "powerball")
                alertDialog.dismiss()
            }
            alertDialog.show()
        }
        megaMillions!!.setOnClickListener {
            val dialogBuilder = AlertDialog.Builder(activity!!)
            // ...Irrelevant code for customizing the buttons and title
            val inflater = activity!!.layoutInflater
            val dialogView = inflater.inflate(R.layout.ticket_selector, null)
            val one = dialogView.findViewById<View>(R.id.one) as NumberPicker
            one.minValue = 1
            one.maxValue = 69
            val two = dialogView.findViewById<View>(R.id.two) as NumberPicker
            two.minValue = 1
            two.maxValue = 69
            val three = dialogView.findViewById<View>(R.id.three) as NumberPicker
            three.minValue = 1
            three.maxValue = 69
            val four = dialogView.findViewById<View>(R.id.four) as NumberPicker
            four.minValue = 1
            four.maxValue = 69
            val five = dialogView.findViewById<View>(R.id.five) as NumberPicker
            five.minValue = 1
            five.maxValue = 69
            val six = dialogView.findViewById<View>(R.id.six) as NumberPicker
            six.minValue = 1
            six.maxValue = 26
            six.setBackgroundColor(Color.YELLOW)
            val group = dialogView.findViewById<View>(R.id.group) as RadioGroup
            dialogBuilder.setView(dialogView)
            val alertDialog = dialogBuilder.create()
            alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "Cancel", null) { dialog, which -> alertDialog.dismiss() }
            alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "Add", null) { dialog, which -> //megaNumbers.clear();
                //megaNumbers = SharedPrefHelper.getMyTickets(getActivity(),"megamillions");
                // megaNumbers = new ArrayList<MyTicket>(megaAdapter.mDataset);
                val ticket = one.value.toString() + " " + two.value + " " + three.value +
                        " " + four.value + " " + five.value + " " + six.value
                val mul = group.checkedRadioButtonId
                val but = dialogView.findViewById<RadioButton>(mul)
                val multi = but.text.toString()
                var ticketMultiplier = false
                if (multi == "Yes") {
                    ticketMultiplier = true
                }
                val ticket1 = MyTicket()
                ticket1.ticket = ticket
                ticket1.multi = ticketMultiplier
                megaNumbers!!.add(ticket1)
                megaAdapter!!.notifyItemInserted(megaNumbers!!.size - 1)
                //megaAdapter.notifyDataSetChanged();
                SharedPrefHelper.setMyTickets(this.context!!, megaNumbers, "megamillions")
                alertDialog.dismiss()
            }
            alertDialog.show()
        }
    }

    companion object {
        fun newInstance(): MyNumbers {
            return MyNumbers()
        }
    }
}